### PR TITLE
Log errors in default logger + refactor

### LINKED ISSUE #50: Default logger should also log errors to std::cerr
Citation from README.md
> In default configuration it logs only warnings and errors to std::cerr.

But `Default` logger logs only warnings, errors are omitted.

### PR DESCRIPTION
**Pull request by OpatrilPeter**
Some minor formating may still be in order, but the logic should be ok.
Fixes https://github.com/seznam/SuperiorMySqlpp/issues/50

Edit: after first internal review, this should be ready.

### CODE REVIEW

File: include/superior_mysqlpp/logging.hpp [hunk 1]
```diff
@@ -63,6 +38,27 @@ namespace SuperiorMySqlpp {
          * so you shall either implement all logging functions or only those you want.
          */
 
+        namespace detail {
+
+            /**
+             * Global variable guaranteeing single access to stderr from standard implementation of loggers
+             * @remark Conceptally should be hidden in #logStderr function, but that would lead to multiple instances.
+             */
+            std::atomic_flag stderrLogWriteLock{};
```
Comment (OpatrilPeter) [inline include/superior_mysqlpp/logging.hpp]: Global variables are usually warning sing, so I'm pointing it out. My previous solution passed `atomic_lock`s around as they were used previously, this one leads to less boilerplate with no obvious disadvantage for me.

Comment (OpatrilPeter) [reply include/superior_mysqlpp/logging.hpp]: My bad, this rewrite looks nicer then original approach, but global variables in header-only lib are no-no, of course.

File: include/superior_mysqlpp/logging.hpp [hunk 2]
```diff
@@ -12,40 +12,31 @@
...
-    template<typename Stream, typename T, typename... Args>
-    void stringify_helper(Stream& stream, T&& value, Args&&... args)
-    {
-        stringify_helper(stream, value);
-        stringify_helper(stream, std::forward<Args>(args)...);
-    }
+        /**
+         * Helper function printing its arguments into std::cerr via operator <<
+         * @remark Parameters are perfect forwarded and are followed by implicit std::endl.
+         * @param args
```
Comment (smradCZ) [inline include/superior_mysqlpp/logging.hpp]: Please fill up all parameters to doxygen comment

File: include/superior_mysqlpp/utils.hpp [hunk 3]
```diff
@@ -54,4 +54,25 @@ namespace SuperiorMySqlpp
         constexpr std::size_t tuple_size = std::tuple_size<Tuple>::value;
         return detail::invokeViaTupleImpl(f, t, std::make_index_sequence<tuple_size>{});
     }
+
+    /**
+     * Writes variable number of arguments into arbitrary std(-like) stream.
```
Comment (smradCZ) [inline include/superior_mysqlpp/utils.hpp:59]: It writes *exactly one* argument

File: include/superior_mysqlpp/utils.hpp [hunk 4]
```diff
@@ -54,4 +54,25 @@ namespace SuperiorMySqlpp
...
+     * @param arg Perfectly forwarded argument
+     * @return Stream reference
+     */
+    template <typename Stream, typename Arg>
+    Stream &streamify(Stream &stream, Arg &&arg)
+    {
+        return stream << std::forward<Arg>(arg);
+    }
+
+    /**
```
Comment (smradCZ) [inline include/superior_mysqlpp/utils.hpp:70]: Please fill full doxydoc here to keep whole class documented.

**Review by smradCZ — CHANGES_REQUESTED** [review]
Good point, it's much better to have only one definition of log messages on one place.
There are some small mistakes in doxygen comments, please fix them and then we can merge your changes.
Thanks a lot

File: include/superior_mysqlpp/utils.hpp (hunk 3 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/utils.hpp:59]: This is by design, goal is that overloads should share the documentation. Apparently, `@overload` was needed for that, judging by other examples in this codebase it wasn't clear.

File: include/superior_mysqlpp/logging.hpp (hunk 2 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/logging.hpp]: Well spotted, forgot about lock being added in last refactor.

**Review by smradCZ — APPROVED** [review]

**Pull request merged**